core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,25 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Generic test that first adds a "boost" field to the otherwise randomly created query and
+     * then replaces the field with an unknown field. This should trigger ParsingException that
+     * points out the unknown field.
+     * Queries that allow arbitrary field names at this level need to override this test so it
+     * doesn't get executed for those specific tests.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        testQuery.boost(1.0f);
+        String queryAsString = testQuery.toString().replace("boost", "bogusField");
```

Comment:
regardless of where boost is, isn't is ok if we replace only when there's only one occurrence of it in the string query? Otherwise we skip the test? I think it's a best effort that should be ok 99% of the cases... unless I am missing something